### PR TITLE
Various security issues

### PR DESCRIPTION
**Pull request by ashu555**
Fixes existing security issues in the system.
Refer https://docs.google.com/document/d/1cM_RbMkb6ckJC8U8e_FRz2lnFTNIVJXlK-eOmnfjV5s/edit

### CODE REVIEW

**Comment by bindu95** [comment]
@ashu555 , still am able to update the API keys as client by changing the method


1 | AUTO UPDATE | http://localhost/agora-invoicing-community/public/updatedomainCheckDetails?status=1
-- | -- | --
2 | RECAPTCHA | http://localhost/agora-invoicing-community/public/captchaDetails?status=0
3 | MSG91 | http://localhost/agora-invoicing-community/public/updatemobileDetails?status=0
4 | MAILCHIMP | http://localhost/agora-invoicing-community/public/updateMailchimpDetails?status=0
5 | SHOW TERMS ON REGISTRATION PAGE | http://localhost/agora-invoicing-community/public/updateTermsDetails?status=0
6 | EMAIL VERIFICATION | http://localhost/agora-invoicing-community/public/updateemailDetails?status=0
7 | TWITTER | http://localhost/agora-invoicing-community/public/updatetwitterDetails?status=0
8 | ZOHO CRM | http://localhost/agora-invoicing-community/public/updatezohoDetails?status=1&zoho_key=a0f6d2d14faf714e787802bb5ade1551
9 | PIPEDRIVE | http://localhost/agora-invoicing-community/public/updatepipedriveDetails?status=1&pipedrive_key=05b5423d4bf22112fb19bac9f8f2fe18a6e7a549
10 | RAZORPAY | localhost/agora-invoicing-community/public/updaterzpDetails?status=1&rzp_key=rzp_test_smpCLxN6RfoHuo&rzp_secret=EsOfGLnR9Nk6FOxyhh3ITzFk&apilayer_key=1af85deb04dd0c538c06c5c005ef73cf
11 | Don’t allow ip/Domain | http://localhost/agora-invoicing-community/public/updatedomainCheckDetails?status=1
12 | Auto PHP licenser | http://localhost/agora-invoicing-community/public/licenseDetails?status=1&license_api_secret=5hDuaXuTh9gTLfPA&license_api_url=https://license.faveohelpdesk.com/apl_api/api.php

**Comment by bindu95** [comment]
Adding the gif for the steps I did
CE6MvkrU4k

**Comment by bindu95** [comment]
Removing the gif as discussed(open source and anyone can access) but please check the places mentioned as well as a complete system for the same

**Comment by bindu95** [comment]
From next time before starting there should be another step of code reviewing. Once code approved then we can start testing @ashu555 , please check with concern developers for code reviewing as well.

File: app/Http/Controllers/Payment/CurrencyController.php [hunk 1]
```diff
@@ -103,11 +103,11 @@ public function getButtonColor($id)
         }
         $currency = Currency::where('id', $id)->pluck('dashboard_currency')->first();
         if ($currency == 1) {
-            return '<a href='.url('dashboard-currency/'.$id)." 
+            return '<a href='.action('Payment\CurrencyController@setDashboardCurrency', [$id])."
```
Comment (mverma16) [inline app/Http/Controllers/Payment/CurrencyController.php:106]: what difference will it make in href link? In fresh installation not working throwing this response
```
{"draw":1,"recordsTotal":161,"recordsFiltered":0,"data":[],"error":"Exception Message:\n\nAction App\\Http\\Controllers\\Payment\\CurrencyController@setDashboardCurrency not defined."}
```

File: resources/views/themes/default1/common/plugins.blade.php [hunk 2]
```diff
@@ -140,5 +140,14 @@
                 $('.loader').css('display', 'block');
             },
         });
+
+      
+    </script>
+    <script>
+           $(document).ready(function(){
+              $('#deactivate').click(function(){
+            alert('dsf');
```
Comment (mverma16) [inline resources/views/themes/default1/common/plugins.blade.php:149]: Do you really want to show this alert?

File: resources/views/themes/default1/front/auth/login-register.blade.php [hunk 3]
```diff
@@ -669,7 +669,7 @@ function verifyBySendOtp() {
                 };
                 $.ajax({
                     url: '{{url('otp/verify')}}',
-                    type: 'GET',
+                    type: 'post',
```
Comment (mverma16) [inline resources/views/themes/default1/front/auth/login-register.blade.php:672]: Keep consistency throughout, either use POST or post.

File: app/Http/Controllers/Payment/CurrencyController.php (hunk 1 above)
Comment (ashu555) [reply app/Http/Controllers/Payment/CurrencyController.php:106]: Fixed

File: resources/views/themes/default1/common/plugins.blade.php (hunk 2 above)
Comment (ashu555) [reply resources/views/themes/default1/common/plugins.blade.php:149]: Fixed

File: resources/views/themes/default1/front/auth/login-register.blade.php (hunk 3 above)
Comment (ashu555) [reply resources/views/themes/default1/front/auth/login-register.blade.php:672]: fixed

File: resources/views/themes/default1/layouts/front/master.blade.php [hunk 4]
```diff
@@ -543,9 +542,9 @@
       var email = $('#newsletterEmail').val();
       $('#mailchimp-subscription').html("<i class='fa fa-circle-o-notch fa-spin fa-1x fa-fw'></i>Please Wait...");
       $.ajax({
-        type: 'GET',
+        type: 'POST',
         url: '{{url("mail-chimp/subcribe")}}',
```
Comment (mverma16) [inline resources/views/themes/default1/layouts/front/master.blade.php:546]: Does this route not processed by VerfifyCSRF middleware?

Comment (ashu555) [reply resources/views/themes/default1/layouts/front/master.blade.php:546]: Fixed

File: resources/views/themes/default1/layouts/front/master.blade.php [hunk 5]
```diff
@@ -603,7 +602,7 @@
 
      function removeItem(id) {
             $.ajax({
-              type: "GET",
+              type: "POST",
```
Comment (mverma16) [inline resources/views/themes/default1/layouts/front/master.blade.php:605]: same goes here

Comment (ashu555) [reply resources/views/themes/default1/layouts/front/master.blade.php:605]: fixed

**Comment by joeleonjr** [comment]
Hi! I'm a security researcher @TruffleSecurity. I found a valid API key leaked in a comment in this Issue. I highly recommend rotating the API key immediately (not just deleting the comment). Here's a blog post I wrote on remediating these types of vulnerabilities: https://trufflesecurity.com/blog/remediate-leaked-api-keys-with-key-rotation.<br><br>If you can't find the leaked API key, you can use the following [TruffleHog](https://github.com/trufflesecurity/trufflehog) command to automatically scan for it:<br>`trufflehog github --repo https://github.com/<your_username>/<your_reponame> --issue-comments --only-verified`<br><br>If it's not clear where/what the security issue is, please let me know. I'd be happy to help. Please note that the API key might be in the past edit of a comment.<br><br><br><br>(cc @bindu95)

**Pull request merged**